### PR TITLE
Add subject with a namespace.

### LINKED ISSUE #2140: "wskadmin user get guest" broken
See this commit: https://github.com/openwhisk/openwhisk/commit/7b5a64481aedfd060f3ccaab72a2b6197de47df2#commitcomment-21755818

### PR DESCRIPTION
**Pull request by rabbah**
Closes #2140.

This reverts a change from this commit https://github.com/apache/incubator-openwhisk/commit/7b5a64481aedfd060f3ccaab72a2b6197de47df2#commitcomment-21755818.

### CODE REVIEW

**Comment by csantanapr** [comment]
@cbickel can you take a look at this, should I merge this?

**Comment by rabbah** [comment]
@csantanapr I have to pg this.

**Comment by rabbah** [comment]
Adding @markusthoemmes as the scope has expanded to remove dead code.

**Comment by rabbah** [comment]
pg2/1654 🔵

**Comment by csantanapr** [comment]
@rabbah could address the conflict, and I will merge

**Comment by rabbah** [comment]
I rebase and added a new test. New pg running.

**Comment by rabbah** [comment]
pg1/1724 🔵

File: tests/src/test/scala/system/basic/WskBasicTests.scala [hunk 1]
```diff
@@ -173,12 +171,10 @@ class WskBasicTests
                 (pkg, _) => pkg.create(name, parameters = paramInput)
             }
 
-            val expectedParam = JsObject(
-                "payload" -> JsString("test"))
-
-            val ns_regex_list = wsk.namespace.list().stdout.trim.replace('\n', '|')
+            val expectedParam = JsObject("payload" -> JsString("test"))
+            val ns = s""""${wsk.namespace.whois()}""""
```
Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskBasicTests.scala]: Is this adding quotes around the string?

File: tests/src/test/scala/system/basic/WskBasicTests.scala [hunk 2]
```diff
@@ -817,8 +788,10 @@ class WskBasicTests
     behavior of "Wsk Namespace CLI"
 
     it should "return a list of exactly one namespace" in {
-        wsk.namespace.list().
-            stdout.lines should have size 2 // headline + namespace
+        val lines = wsk.namespace.list().stdout.lines.toSeq
+        assert(lines.length == 2)
```
Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskBasicTests.scala]: `lines should have size 2`

File: tests/src/test/scala/system/basic/WskBasicTests.scala [hunk 3]
```diff
@@ -817,8 +788,10 @@ class WskBasicTests
...
+        assert(lines(0) == "namespaces")
```
Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskBasicTests.scala]: `lines.head shouldBe "namespaces"`

File: tests/src/test/scala/whisk/core/connector/test/MessageFeedTests.scala [hunk 4]
```diff
@@ -56,7 +56,11 @@ class MessageFeedTests
     val system = ActorSystem("MessageFeedTestSystem")
     val actorsToDestroyAfterEach: Buffer[ActorRef] = Buffer.empty
 
-    override def afterEach() = actorsToDestroyAfterEach.foreach { _ ! PoisonPill }
+    override def afterEach() = {
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/connector/test/MessageFeedTests.scala]: @markusthoemmes fyi

**Comment by rabbah** [comment]
Rebased to remove conflicts.

**Comment by rabbah** [comment]
PG1/1789 and PG4/530 both blue after rebase and conflict resolution.

File: tests/src/test/scala/system/basic/WskBasicTests.scala [hunk 5]
```diff
@@ -735,50 +711,45 @@ class WskBasicTests
...
-            wsk.rule.get(ruleName, fieldFilter = Some("status")).stdout should include(s"""$successMsg status\n"active"""")
-            val trigger = wsk.rule.get(ruleName, fieldFilter = Some("trigger")).stdout
-            trigger should include regex (s"""$successMsg trigger\n""")
-            trigger should include(triggerName)
-            trigger should not include (actionName)
-            val action = wsk.rule.get(ruleName, fieldFilter = Some("action")).stdout
-            action should include regex (s"""$successMsg action\n""")
-            action should include(actionName)
-            action should not include (triggerName)
+        val ns = s""""${wsk.namespace.whois()}""""
```
Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskBasicTests.scala]: Can be just `wsk.namespace.whois()`

File: tests/src/test/scala/system/basic/WskBasicTests.scala [hunk 6]
```diff
@@ -570,73 +556,63 @@ class WskBasicTests
...
-
-            val expectedParam = JsObject(
-                "payload" -> JsString("test"))
+        (wp, assetHelper) => assetHelper.withCleaner(wsk.trigger, name) {
+            (trigger, _) => trigger.create(name, parameters = paramInput)
+        }
 
-            val ns_regex_list = wsk.namespace.list().stdout.trim.replace('\n', '|')
+        val expectedParam = JsObject("payload" -> JsString("test"))
+        val ns = s""""${wsk.namespace.whois()}""""
```
Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskBasicTests.scala]: same.

File: tests/src/test/scala/system/basic/WskBasicTests.scala [hunk 7]
```diff
@@ -817,8 +788,10 @@ class WskBasicTests
     behavior of "Wsk Namespace CLI"
 
     it should "return a list of exactly one namespace" in {
-        wsk.namespace.list().
-            stdout.lines should have size 2 // headline + namespace
+        val lines = wsk.namespace.list().stdout.lines.toSeq
+        lines.length shouldBe 2
```
Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskBasicTests.scala]: lines should have size 2

File: tests/src/test/scala/system/basic/WskBasicTests.scala [hunk 8]
```diff
@@ -846,13 +819,12 @@ class WskBasicTests
                     trigger.create(name)
             }
 
-            val ns_regex_list = wsk.namespace.list().stdout.trim.replace('\n', '|')
-
+            val ns = s""""${wsk.namespace.whois()}""""
```
Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskBasicTests.scala]: same.

**Pull request merged**